src/api/test/unit/project_test.rb

```diff
@@ -18,18 +18,47 @@ def test_maintained_project_names
     assert_equal ['A', 'B', 'C'], project.maintained_project_names
   end
 
+  # TODO: used by bento. Remove when dropping old UI.
+  # The unit tests need to be adapted since they depend on this
+  # Returns a hash of arrays, sorted by repository.
+  # The arrays contain Flag objects of type, sorted by architecture.
+  # Like:
+  # {all: [Flag, Flag-i586, Flag-x86_64], 13.2: [Flag, Flag-i585, Flag-x86_64]}
+  def get_flags(prj_or_pkg, flag_type, repository_names, architectures)
+    the_flags = {}
+
+    # [nil] is a placeholder for "all" repositories
+    [nil].concat(repository_names).each do |repository|
+      the_flags[repository] = []
+      # [nil] is a placeholder for "all" architectures
+      [nil].concat(architectures).each do |architecture|
+        architecture_id = architecture ? architecture.id : nil
+        flag = prj_or_pkg.flags.where(flag: flag_type).where(repo: repository).where(architecture_id: architecture_id).first
+        # If there is no flag create a temporary one.
+        unless flag
+          flag = prj_or_pkg.flags.new(flag: flag_type, repo: repository, architecture: architecture)
+          flag.status = flag.effective_status
+        end
+        the_flags[repository] << flag
+      end
+    end
+    the_flags['all'] = the_flags.delete(nil)
+
+    the_flags
+  end
+
   # TODO: bento_only
   def _project_flags(project, flag)
     architectures = Architecture.where(id: project.repository_architectures.select(:architecture_id)).reorder('name').distinct
     repository_names = project.repositories.pluck(:name)
-    project.get_flags(flag, repository_names, architectures)
+    get_flags(project, flag, repository_names, architectures)
```

Comment:
get rid of the complete test instead. There is no point in testing a function that only exists in the test itself.

Reply:
@coolo I thought the specified_flags rspec test is not covering all the scenarios, but yeah I looked again, and it seems its covering the inheritance of the flags (even though a bit simpler). So I will remove all this stuff.